### ISSUE TITLE
v0.2.1 release artifacts for Linux x86_64 contain 0.2.0 binaries

### ISSUE THREAD

**Issue by joyshmitz** [Dicklesworthstone/mcp_agent_mail_rust#36]
## Summary
Release tag `v0.2.1` currently serves Linux x86_64 binaries that report `0.2.0`.

This blocks actual upgrade validation because installer and manual download both end up with old-version binaries.

## Reproduction
```bash
VERSION=v0.2.1
BASE="https://github.com/Dicklesworthstone/mcp_agent_mail_rust/releases/download/${VERSION}"

curl -fsSL "$BASE/mcp-agent-mail-x86_64-unknown-linux-gnu.tar.gz" -o /tmp/am.tar.gz
mkdir -p /tmp/am-v021 && tar -xf /tmp/am.tar.gz -C /tmp/am-v021
/tmp/am-v021/am --version
/tmp/am-v021/mcp-agent-mail --version
```

Observed on **March 3, 2026**:
```text
am 0.2.0
mcp-agent-mail 0.2.0
```

Same result when downloading `mcp-agent-mail-x86_64-unknown-linux-gnu.tar.xz`.

## Expected
`v0.2.1` release artifacts should report `0.2.1`.

## Suggested fix
Rebuild/re-upload release binaries for `v0.2.1` (or publish `v0.2.2`) ensuring Cargo package version baked into binaries matches the release tag.
